src/NuGet.Core/NuGet.ProjectModel/PackageSpecOperations.cs

```diff
@@ -26,7 +26,7 @@ public static void AddOrUpdateDependency(PackageSpec spec, PackageDependency dep
 
             if (!existing.Any())
             {
```

Comment:
All the modifications in this file are already in review based on @jeffkl  PR (https://github.com/NuGet/NuGet.Client/pull/4642) so I will remove these changes once his PR is merged into dev.

Reply:
can you rebase that way the changeset is minimized and a bit easier to review?

Reply:
Agreed, this has been resolved.